stroyprombeton/tests/tests_views.py

```diff
@@ -741,13 +741,35 @@ class ProductPrice(TestCase):
 
     fixtures = ['dump.json']
 
-    def test_price_list(self):
-        """Context for pdf generation should include Category and Products."""
-        self.response = self.client.get('/gbi/categories/1/pdf/')
+    def setUp(self):
+        self.category = models.CategoryPage.objects.filter(level=0).first()
+        self.response = self.client.get(
+            reverse('product_pdf', args=(self.category.id,))
+        )
+        self.context = self.response.context
+
+    def test_content_type(self):
+        self.assertEqual(self.response['Content-Type'], 'application/pdf')
+
+    def test_category_name(self):
+        self.assertEqual(self.context['category'].name, self.category.name)
+
+    def test_products(self):
+        self.assertIsInstance(self.context['products'], models.OptionQuerySet)
+        self.assertGreater(self.context['products'].count(), 100)
+
+    def test_tags_group(self):
+        # it is grouped
+        for group, tags in self.context['group_tags_pairs']:
+            for tag_ in tags:
+                self.assertEqual(group, tag_.group)
+
+        # it is products' tags
+        _, grouped_tags = zip(*self.context['group_tags_pairs'])
```

Comment:
```suggestion
        grouped_tags = [tags for _, tags in self.context['group_tags_pairs']
```
Would be more readable in my taste.

or even the whole expression
```
tags = set([t for tags for _, tags in self.context['group_tags_pairs'] for t in tags])
```

Up to you